web/src/pages/api/public/metrics/daily.ts

```diff
@@ -47,53 +47,43 @@ export default withMiddlewares({
               SUM(o.completion_tokens) "outputUsage",
               SUM(o.total_tokens) "totalUsage",
               COALESCE(SUM(o.calculated_total_cost), 0)::DOUBLE PRECISION as "totalCost"
-            FROM
-              traces t
-            LEFT JOIN observations_view o ON o.trace_id = t.id AND o.project_id = t.project_id
+            FROM traces t
+            LEFT JOIN observations_view o
+            ON o.trace_id = t.id
+            AND o.project_id = t.project_id
             WHERE o.start_time IS NOT NULL
+              AND o.project_id = ${auth.scope.projectId}
               AND t.project_id = ${auth.scope.projectId}
               ${traceNameCondition}
               ${userCondition}
               ${tagsCondition}
               ${fromTimestampCondition}
               ${toTimestampCondition}
-            GROUP BY
-              1,
-              2
-            ORDER BY
-              1,
-              2
+            GROUP BY 1, 2
           ),
           daily_model_usage AS (
             SELECT
               "date",
-              json_agg(json_build_object('model',
-                  model,
-                  'inputUsage',
-                  "inputUsage",
-                  'outputUsage',
-                  "outputUsage",
-                  'totalUsage',
-                  "totalUsage",
-                  'totalCost',
-                  "totalCost",
-                  'countObservations',
-                  "countObservations",
-                  'countTraces',
-                  "countTraces")) daily_usage_json
+              sum("countObservations")::integer "countObservations",
+              sum("totalCost")::DOUBLE PRECISION "totalCost",      
+              json_agg(json_build_object(
+                'model', "model",
+                'inputUsage', "inputUsage",
+                'outputUsage', "outputUsage",
+                'totalUsage', "totalUsage",
+                'totalCost', "totalCost",
+                'countObservations', "countObservations",
+                'countTraces', "countTraces"
+              )) daily_usage_json
             FROM
               model_usage
-            GROUP BY
-              1
+            GROUP BY 1
           ),
-          daily_stats AS (
+          trace_usage AS (
             SELECT
               DATE_TRUNC('DAY', t.timestamp) "date",
-              count(distinct t.id)::integer count_traces,
-              count(distinct o.id)::integer count_observations,
-              SUM(o.calculated_total_cost)::DOUBLE PRECISION total_cost
+              count(distinct t.id)::integer as "countTraces"
```

Comment:
Unfortunately, I couldn't drop the whole query as the aggregation in the first query would double count traces if we group by model. But we can still reduce this query to avoid the join with observations which should be a net-positive.